src/libraries/Hooks.sol

```diff
@@ -148,6 +148,25 @@ library Hooks {
         (, delta) = abi.decode(result, (bytes4, int256));
     }
 
+    /// @notice performs a hook call using the given calldata on the given hook
+    /// @return delta The delta returned by the hook
+    /// @return fee The fee returned by the hook
+    function callHookWithReturnDeltaAndFee(IHooks self, bytes memory data, bool parseReturn, bool parseFee)
+        internal
+        returns (int256 delta, uint24 fee)
+    {
+        bytes memory result = callHook(self, data);
+        (, delta, fee) = abi.decode(result, (bytes4, int256, uint24));
+
+        if (!parseReturn) {
```

Comment:
I think now that there is more than 1 thing that is returned and needs to be parsed we can change this var name to parseDelta